docs/key-issues/gpus.md

```diff
@@ -180,3 +204,85 @@ Potential solutions on the PyPI side include:
 - make an environment marker or selector package approach work,
 - improve interoperability with other package managers, in order to be able to
   declare a dependency on a CUDA or ROCm version as externally provided,
+
+
+## Additional Notes on CUDA Compatibility
+
+Compatibility across CUDA versions is a common problem with numerous pitfalls to be aware of.
+There are three primary components of CUDA:
+
+1. The CUDA Toolkit (CTK): This component includes the CUDA runtime library
+   (libcudart.so) along with a range of other libraries and tools including
+   math libraries like cuBlas. libcudart.so and a few other core headers and
+   libraries compose the bare minimum required to compile CUDA code.
+2. The user-mode driver: This is the libcuda.so library. This library is
+   required to actually run CUDA code.
+3. The kernel-mode driver: The nvidia.ko file. This constitutes what is
+   typically considered a "driver" in common parlance when referring to other
+   peripherals connected to a computer.
+
+CUDA developers and users generally do not need to be aware of the kernel-mode driver.
```

Comment:
I would argue that CUDA forward compat is only relevant for HPC _admins_, whereas users and developers do not usually care about it. I would like to leave forward compat out of this discussion altogether to avoid muddying the waters.

Reply:
Following up here: https://github.com/pypackaging-native/pypackaging-native/pull/36#discussion_r1239276086

Reply:
Resolved on the other thread.